library/core/src/ptr/non_null.rs

```diff
@@ -1416,6 +1418,8 @@ impl<T: ?Sized> NonNull<T> {
     #[must_use]
     #[unstable(feature = "pointer_is_aligned_to", issue = "96284")]
     #[rustc_const_unstable(feature = "const_pointer_is_aligned", issue = "104203")]
+    #[requires(align.is_power_of_two() && align > 0)] // Ensure alignment is a power of two and not zero
```

Comment:
This pre-condition is also not needed. As a rule of thumb, safe methods don't really need pre-conditions in order to prove safety.

Reply:
removed